temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java

```diff
@@ -59,64 +60,91 @@
  */
 public class TestWorkflowRule implements TestRule {
 
-  public Timeout globalTimeout;
+  private static final long DEFAULT_TEST_TIMEOUT_SECONDS = 10;
 
+  // Only enable when USE_DOCKER_SERVICE is true
+  private static final Boolean useExternalService =
+      Boolean.parseBoolean(System.getenv("USE_DOCKER_SERVICE"));
```

Comment:
Just a thought: this rule is used by clients that would be testing more than just Temporal, so in that context, `USE_DOCKER_SERVICE` might not be very meaningful.

Reply:
I agree. Let's remove `useExternalService` and `temporalServiceAddress` initialization from here and keep only in SDKTestWorkflowRule.

Reply:
Just to clarify, we are talking about removing environment variable and not about removing ability to run against external service per se?

Reply:
Correct